sycl/source/detail/device_impl.cpp

```diff
@@ -284,6 +284,20 @@ bool device_impl::has(aspect Aspect) const {
                MDevice, PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE,
                sizeof(pi_device_type), &device_type,
                &return_size) == PI_SUCCESS;
+  case aspect::ext_intel_device_info_uuid: {
+    auto Result = getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+        MDevice, PI_DEVICE_INFO_UUID, 0, nullptr, &return_size);
+    if (Result != PI_SUCCESS) {
+      return false;
+    }
+
+    assert(return_size <= 16);
+    std::byte uuid[16];
+
+    return getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+               MDevice, PI_DEVICE_INFO_UUID, 16 * sizeof(std::byte), uuid,
```

Comment:
`std::byte` needs to be either replaced or guarded here as well.

Reply:
This file is part of the runtime implementation, not a header file that might be compiled into the user's application.  I'm just curious why we cannot use C++17 features inside our runtime implementation.

Reply:
Sorry, you are right, no need to guard in the library.